### ISSUE TITLE
Launched applications become separate from the pinned applications

### ISSUE THREAD

**Issue by YYutopian** [damentz/liquorix-package#156]
Whenever I'm on the Liquorix kernel, it seems as though whenever I launch an application on **Ubuntu 24.04 Noble Numbat**, running on **GNOME 45.3**, the app launches separately from the pinned icon. On Xorg, it's a duplicate of the icon, whereas with Wayland, it is a general gear icon instead.  This can be seen in the Firefox application(tested on Snap and Pacstall), among a few others. I am not sure if this issue applies to other desktop environments.  Thank you for reading this!
![waylandversion](https://github.com/damentz/liquorix-package/assets/157938460/caef4807-bfd5-49f9-bf04-33a809201237)

**Comment by damentz** [comment]
Have you tried opening an issue with the Gnome organization for this behavior?  Even if Liquorix is the reason for this issue, we'll probably need a developer's help to figure out what interaction with the kernel may be causing this.  Once identified, we can figure out how to fix it.

There's also a chance this is a bug that Liquorix is exploiting and needs to be fixed regardless, the same way some specific less well maintained drivers on Liquorix behave differently than when running mainline CFS/EEVDF.

**Comment by YYutopian** [comment]
Let me see what they say about the issue and I'll let you know what they respond with.

As for additional info, someone I spoke with suggested that I use the **printenv**  command in order to gather more clues as to what may be going on. Here are the outputs I've gotten.


[LinuxEnvironment.txt](https://github.com/damentz/liquorix-package/files/14077741/LinuxEnvironment.txt)


[LiquorixEnvironment.txt](https://github.com/damentz/liquorix-package/files/14077742/LiquorixEnvironment.txt)

**Comment by YYutopian** [comment]
According to one of the GNOME maintainers, considering that this is happening on only certain apps, it is most likely an app/Ubuntu-specific issue, as Ubuntu most likely changed the snap app-id format without providing an update to the corresponding code in mutter." I still have little idea as to why it's specifically happening in Liquorix, though.

**Comment by damentz** [comment]
> I still have little idea as to why it's specifically happening in Liquorix, though.

Liquorix is more likely to trigger race conditions due to its usage of PDS over EEVDF.  PDS aggressively runs tasks in parallel on all available idle cores, leading to some code running out of order if it's been programmed incorrectly.  This was a common problem with both BFS and MuQSS back in the day.

Although, this problem was exclusive to kernel drivers, it's bizarre that anything like that is changing the behavior of user space like gnome.  Are you sure you went back and forth and verified?  It's also important to compare like for like with base kernels. A 6.6 stock kernel compared with 6.7 Liquorix isn't fair, since the bugs you're seeing may also exist in 6.7 mainline.

**Comment by YYutopian** [comment]
You've made a really fair point in that I should have compared between equivalent kernels. I went ahead and installed Linux 6.7.2 mainline(I think that's the equivalent kernel), and as it turns out, the bug with the icons isn't there. However, I have noticed that on the mainline 6.7.2 kernel, I believe my sound card didn't get detected and  as a result there is no audio output. 

Thank you for the explanation, by the way!
